### DISCUSSION TITLE
How to fix EDID / Handshake problem?

### DISCUSSION THREAD

**Discussion by ghost** [awawa-dev/HyperHDR#939] [Q&A] [answered]
Hello everyone,

I recently setup my ambilight with HyperHDR and while the software is wonderful, I ran into some hardware issues.

I have a [FeinTech VSP01202](https://feintech.eu/en/products/vsp01202-hdmi-2-0-splitter-1x2-mit-audio-edid-management) splitter which supports 4k HDR. It also has a scaler for 4k to 1080p. I thought simply hooking up my AndroidTV to the splitter, and then to my TV and to the grabber would be sufficient.
However, I ran into some issues when using the splitter in combination with the USB grabber (lsusb confirms its a MS2109, which I got for [7€ from Amazon](https://www.amazon.de/Videoaufnahmekarte-Capture-Streaming-Rekorder-Live-Streaming-hdmi-USB/dp/B0B7DSPPNX/ref=sr_1_6?keywords=capture+card)). Whenever I connect the grabber to either of the splitter outputs, I get a signal for 3-5 seconds, before it turns black. This is the the case for HyperHDR Live video, and also capturing the grabber image with OBS with my laptop. 

It does not matter which Splitter ouput I connect to the grabber, nor does it matter if the TV is connected simultaneously to either of the outputs: It is always the same behaviour. A signal for 3-5s, sometimes some flickering, and then no live signal. I also have a [video demonstration of the Live view.](https://imgur.com/a/XBfXZkX)

I figured there must be some handshake issue between the splitter and specifically the grabber. If I hook up the AndroidTV directly to the grabber, I do get a proper signal, so it is not damaged. Nor is the splitter, as my TV can receive proper 4k HDR.

Unfortunately, I have no idea how to approach this and try to fix this issue. I thought that perhaps the grabber's max input resolution is capped at 1080p30, so I have ordered a different capture card that supposedly accepts 4k60p at the input. I will see if that changes anything when I receive it. I have also read similar experiences on the hyperion-forum where people report similar issues.

Is there any way to fix this, or do I have to replace my hardware and see which combination of splitter + grabber does not cause this issue? I am not too well versed with HDMI, so I maybe one of you can guide me towards a solution/ more detailed explanation of this problem.

**Comment by awawa-dev** [comment]
Hi
Such artifacts suggest rather bad HDMI cables or broken HDMI splitter. ms2130 as an upgrade of ms2109 can handle even 1080p120 on the input although it's not recommended (only low bitrate) but 1080p60 with a support from HDMI splitter should work. Do you have undervolltage detected warning on the main HyperHDR page?

**Reply by ghost** [reply]
hey, thanks for the fast response!
I went ahead and checked the HDMI cables individually already, all of them work fine, even connected every HDMI from source directly to the grabber, works fine. As soon as the splitter is in between, signal for a few seconds then black, no matter which output. 
When I connect the grabber, or dis- and enable the instance in HyperHDR, the connected TV also reconnects and displays the image afterwards, whereas the grabber just goes black around the same time.

Edit: Undervoltage is also not detected.

**Reply by awawa-dev** [reply]
Unfortunately I have no more ideas and if you are 100% sure that you have eliminated the cables then the grabber remains. An EDID conflict is very unlikely, i.e. the splitter sets the refresh parameter (not refresh rate, it cannot change it anyway) outside the range acceptable by the grabber... but then it is also the fault of the splitter firmware. Basically, diagnostics unfortunately consist in trying to replace the splitter / grabber / cables.

**Reply by ghost** [reply]
Yeah I thought so too. I have ordered a new grabber, and also the VSP01201. I will try combinations of everything and report my results on here, in case someone else with a similar hardware setup faces similar problem. 
Anyway, I really appreciate your word on this, definitely needed a second opinion.

**Reply by ghost** [reply, marked as answer]
I received the other [USB grabber](https://www.berrybase.de/usb-hdmi-video-grabber-1080p) today. And everything works now, without any problems.
Funnily enough, lsusb shows the same device ID 534d:2109 for both grabbers, but instead of "Macrosilicon USB Video" it is named "Macrosilicon USB3.0 Capture". Either way, 4K HDR on the TV is working and the grabber image looks good when SDR to HDR tone mapping is enabled.

**Reply by awawa-dev** [reply]
> Funnily enough, lsusb shows the same device ID 534d:2109 for both grabbers,

Classic fake USB3.0 grabber. MS2109 is USB2.0 grabber internally, MS2130 is real USB3.0 grabber.
